skfda/misc/metrics/_sdtw_numba.py

```diff
@@ -0,0 +1,42 @@
+import numpy as np
+from numba import njit
+from ...typing._numpy import NDArrayFloat
+
+
+@njit(cache=True, fastmath=True)
+def _soft_min_argmin(
+    x: float,
+    y: float,
+    z: float
+) -> float:
+    """soft-min of three real numbers"""
+
+    min_xyz = np.minimum(x, np.minimum(y, z))
+    nn = np.exp(min_xyz - x)
+    nn += np.exp(min_xyz - y)
+    nn += np.exp(min_xyz - z)
+
+    return min_xyz - np.log(nn)
+
+
+@njit(cache=True)
+def _sdtw(
```

Comment:
Do you know if this is this the same (or similar) dynamic programming algorithm used for Fisher-Rao registration? If it is, we could also use it in that case and remove the dependency on fdasrsf.